packages/accounts/src/policies/simple_threshold.rs

```diff
@@ -0,0 +1,221 @@
+/// * `context_rule` - The context rule for this policy.
+/// * `smart_account` - The address of the smart account.
+///
+/// # Events
+///
+/// * topics - `["simple_policy_enforced", smart_account: Address]`
+/// * data - `[context: Context, authenticated_signers: Vec<Signer>]`
+pub fn enforce(
+    e: &Env,
+    context: &Context,
+    authenticated_signers: &Vec<Signer>,
+    context_rule: &ContextRule,
+    smart_account: &Address,
+) {
+    // Require authorization from the smart_account
+    smart_account.require_auth();
+
+    if can_enforce(e, context, authenticated_signers, context_rule, smart_account) {
+        // emit event
+        e.events().publish(
+            (Symbol::new(e, "simple_policy_enforced"), smart_account),
+            (context.clone(), authenticated_signers.clone()),
+        );
+    }
+}
+
+/// Sets the threshold value for a smart account's simple threshold policy.
+/// Requires authorization from the smart account.
+///
+/// # Arguments
+///
+/// * `e` - Access to the Soroban environment.
+/// * `threshold` - The minimum number of signers required for authorization.
+/// * `context_rule` - The context rule for this policy.
+/// * `smart_account` - The address of the smart account.
+///
+/// # Errors
+///
+/// * [`SimpleThresholdError::InvalidThreshold`] - When threshold is 0 or
+///   exceeds the total number of signers.
+pub fn set_threshold(e: &Env, threshold: u32, context_rule: &ContextRule, smart_account: &Address) {
+    // Require authorization from the smart_account
+    smart_account.require_auth();
+
+    if threshold == 0 || threshold > context_rule.signers.len() {
+        panic_with_error!(e, SimpleThresholdError::InvalidThreshold)
+    }
+
+    e.storage().persistent().set(
+        &SimpleThresholdStorageKey::AccountContext(smart_account.clone(), context_rule.id),
+        &threshold,
+    );
+}
+
+/// Installs the simple threshold policy on a smart account.
+/// Requires authorization from the smart account.
+///
+/// # Arguments
+///
+/// * `e` - Access to the Soroban environment.
+/// * `params` - Installation parameters containing the threshold.
+/// * `context_rule` - The context rule for this policy.
+/// * `smart_account` - The address of the smart account.
+///
+/// # Errors
+///
+/// * [`SimpleThresholdError::InvalidThreshold`] - When threshold is 0 or
+///   exceeds the total number of signers in the context rule.
+pub fn install(
+    e: &Env,
+    params: &SimpleThresholdAccountParams,
+    context_rule: &ContextRule,
+    smart_account: &Address,
+) {
+    // Require authorization from the smart_account
+    smart_account.require_auth();
+
+    if params.threshold == 0 || params.threshold > context_rule.signers.len() {
+        panic_with_error!(e, SimpleThresholdError::InvalidThreshold)
+    }
+
+    e.storage().persistent().set(
+        &SimpleThresholdStorageKey::AccountContext(smart_account.clone(), context_rule.id),
+        &params.threshold,
+    );
+}
```

Comment:
identical function bodies with `set_threshold` (except `params.threshold` vs `threshold`).

Should be possible to re-use the most of the code